x/cronos/types/messages.go

```diff
@@ -171,3 +176,32 @@ func (msg *MsgUpdateTokenMapping) GetSignBytes() []byte {
 	bz := ModuleCdc.MustMarshalJSON(msg)
 	return sdk.MustSortJSON(bz)
 }
+
+func NewMsgUpdateParams(authority string, params Params) *MsgUpdateParams {
+	return &MsgUpdateParams{
+		Authority: authority,
+		Params:    params,
+	}
+}
+
+// GetSigners returns the expected signers for a MsgUpdateParams message.
+func (msg *MsgUpdateParams) GetSigners() []sdk.AccAddress {
+	addr, err := sdk.AccAddressFromBech32(msg.Authority)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{addr}
+}
+
+// ValidateBasic does a sanity check on the provided data.
+func (msg *MsgUpdateParams) ValidateBasic() error {
+	if _, err := sdk.AccAddressFromBech32(msg.Authority); err != nil {
+		return sdkerrors.Wrap(err, "invalid authority address")
+	}
+
+	if err := msg.Params.Validate(); err != nil {
```

Comment:
Do we need validate for ibc related params here?

Reply:
yeah and `Params.Validate()` does what you want.

Reply:
might just fix upgrade test then, complain `IbcCroDenom` not registered

Reply:
how could we check if `IbcCroDenom` is registered or not?

Reply:
seems we miss register for [cronos](https://github.com/adu-crypto/cronos/blob/73f9ebdaf12408e15e163a7fa591ef8c85a166b9/app/app.go#L971) 
```paramsKeeper.Subspace(cronostypes.ModuleName).WithKeyTable(cronostypes.ParamKeyTable())```